build/yamls/externalnode/vm-agent-rbac.yml

```diff
@@ -0,0 +1,102 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: vm-agent
+  namespace: vm-ns # Change the Namespace to where vm-agent is expected to run.
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vm-agent
+rules:
+  # antrea-controller distributes the CA certificate as a ConfigMap named `antrea-ca` in the Antrea deployment Namespace.
+  # vm-agent needs to access `antrea-ca` to connect with antrea-controller.
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+    resourceNames:
+      - antrea-ca
+    verbs:
+      - get
+      - watch
+      - list
+  # This is the content of built-in role kube-system/extension-apiserver-authentication-reader.
+  # But it doesn't have list/watch permission before K8s v1.17.0 so the extension apiserver (vm-agent) will
+  # have permission issue after bumping up apiserver library to a version that supports dynamic authentication.
+  # See https://github.com/kubernetes/kubernetes/pull/85375
+  # To support K8s clusters older than v1.17.0, we grant the required permissions directly instead of relying on
+  # the extension-apiserver-authentication role.
+  - apiGroups:
+      - ""
+    resourceNames:
+      - extension-apiserver-authentication
+    resources:
+      - configmaps
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - crd.antrea.io
+    resourceNames:
+      - vm-1 # Change the ExternalNode name which vm-agent is expected to update.
```

Comment:
Hi @jianjuns @tnqn @wenyingd,

I am not sure if we prefer to specify resourceNames value here or not. 
If the rbac is used by one vm, then it is fine.
But in the cloud setup, the file will be shared for a VPC and the list will be huge.
Maybe we should remove this? I would like to know you idea, thanks!

Reply:
I agree to remove the `resourceNames` here, as it is actually inside ClusterRole definitation, and the ClusterRole actually shares among all VPCs. For per VPC, the ClusterRoleBinding is different as the ClusterRole is expected to bind the ServiceAccount in its namespace.

Reply:
Sounds reasonable to me.

Reply:
Or, we can add a comment saying resourceNames can be used to restrict access to only the current Node.